File: Lista1/ex5/dados/Veiculo.py
import string
from Data import Data

class Veiculo:

    def __init__(self):
        self.__modelo = ""
        self.__marca = ""
        self.__ano = 0
        self.__dataVencimento = Data()

    def getModelo(self):
        return self.__modelo
    
    def setModelo(self, modelo: string):
        self.__modelo = modelo

    def getMarca(self):
        return self.__marca

    def setMarca(self, marca: string):
        self.__marca = marca

    def getAno(self):
        return self.__ano

    def setAno(self, ano: int):
        self.__ano = ano

    def getDataVencimento(self):
        return self.__dataVencimento

    def setDataVencimento(self, data: Data):
        self.__dataVencimento = data

    def __str__(self):
        return 'Modelo: {} \n Marca: {} \n Ano: {} \n Data: {}'.format(self.__modelo, self.__marca, self.__ano, self.__dataVencimento)

File: Listas/Lista1/AluguelDeVeiculosEx1e2/src/dados/Veiculo.java
package dados;
import java.util.Objects;

public class Veiculo {

    private String modelo;
    private String marca;
    private int ano;
    private Data dataVencimento;

    public String getModelo() {
        return modelo;
    }

    public void setModelo(String modelo) {
        this.modelo = modelo;
    }

    public String getMarca() {
        return marca;
    }

    public void setMarca(String marca) {
        this.marca = marca;
    }

    public int getAno() {
        return ano;
    }

    public void setAno(int ano) {
        this.ano = ano;
    }

    public Data getData() {
        return dataVencimento;
    }

    public void setData(Data data) {
        this.dataVencimento = data;
    }

    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Veiculo veiculo = (Veiculo) o;
        return ano == veiculo.ano &&
                Objects.equals(modelo, veiculo.modelo) &&
                Objects.equals(marca, veiculo.marca) && 
                Objects.equals(dataVencimento, veiculo.dataVencimento);
    }

    public String toString() {
        return "Modelo=" + modelo + "\nMarca=" + marca + "\nAno=" + ano + "\n---Data Vencimento---\n" + dataVencimento.toString();
    }
}
